### PR TITLE
[RPA-1381] New approach to repetitive queries

### PR DESCRIPTION
**Pull request by Aeri4a**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

---

I agree to the terms of the RunBotics Contributor License Agreement.

### CODE REVIEW

**Review by preludium — APPROVED** [review]

File: runbotics-orchestrator/src/main/java/com/runbotics/repository/impl/ProcessCustomRepositoryImpl.java [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+        CriteriaQuery<Process> cq = criteriaBuilder.createQuery(Process.class);
+        Root<Process> root =  cq.from(Process.class);
+        Join<Object, Object> user = root.join(Process_.CREATED_BY, JoinType.LEFT);
+        Join<Object, Object> tag = root.join(Process_.TAGS, JoinType.LEFT);
+        cq.select(root);
+        cq.groupBy(root.get(Process_.ID));
+
+        CriteriaQuery<Long> cqCount = criteriaBuilder.createQuery(Long.class);
+        Root<Process> cRoot = cqCount.from(Process.class);
+        Join<Object, Object> cUser = cRoot.join(Process_.CREATED_BY, JoinType.LEFT);
```
Comment (star-in-the-jar) [inline runbotics-orchestrator/src/main/java/com/runbotics/repository/impl/ProcessCustomRepositoryImpl.java:163]: why there're Process and Process_ ?

File: runbotics-orchestrator/src/main/java/com/runbotics/repository/impl/ProcessCustomRepositoryImpl.java [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+            .setParameter(PROCESS_NAME, "%"+name+"%")
+            .setParameter(PROCESS_CREATED_BY_NAME, "%"+createdByName+"%")
+            .setParameter(PROCESS_TAG_NAME, "%"+tagName+"%");
+        if (!hasUserRoleAdmin) query.setParameter(PROCESS_USER, user);
+        if (collectionId != null) query.setParameter(PROCESS_COLLECTION_ID, collectionId);
+
+        TypedQuery<Long> cQuery = entityManager.createQuery(criteriaCountingQuery)
+            .setParameter(PROCESS_NAME, "%"+name+"%")
+            .setParameter(PROCESS_CREATED_BY_NAME, "%"+createdByName+"%")
+            .setParameter(PROCESS_TAG_NAME, "%"+tagName+"%");
```
Comment (star-in-the-jar) [inline runbotics-orchestrator/src/main/java/com/runbotics/repository/impl/ProcessCustomRepositoryImpl.java]: imo "%" + tagName + "%"

File: runbotics-orchestrator/src/main/java/com/runbotics/service/ProcessQueryService.java [hunk 3]
```diff
@@ -136,87 +138,30 @@ public Page<ProcessDTO> findByCriteria(ProcessCriteria criteria, Pageable page)
...
-                    user.getId(),
-                    specification.get(PROCESS_NAME),
-                    specification.get(PROCESS_TAG_NAME),
-                    specification.get(PROCESS_CREATED_BY_NAME),
-                    page
-                )
-                .map(processMapper::toDto);
-        }
+        return processRepository.findBySearch(
+            specification.get(PROCESS_NAME),
```
Comment (star-in-the-jar) [inline runbotics-orchestrator/src/main/java/com/runbotics/service/ProcessQueryService.java:144]: awesome 😎

File: runbotics-orchestrator/src/main/java/com/runbotics/service/ProcessQueryService.java [hunk 4]
```diff
@@ -136,87 +138,30 @@ public Page<ProcessDTO> findByCriteria(ProcessCriteria criteria, Pageable page)
...
-        }
+        return processRepository.findBySearchAndCollection(
+            specification.get(PROCESS_NAME),
+            specification.get(PROCESS_CREATED_BY_NAME),
+            specification.get(PROCESS_TAG_NAME),
+            criteria.getCollectionId() != null ? criteria.getCollectionId().getEquals() : null,
+            user,
+            page
+        ).map(processMapper::toDto);
     }
```
Comment (star-in-the-jar) [inline runbotics-orchestrator/src/main/java/com/runbotics/service/ProcessQueryService.java:165]: noice

**Review by star-in-the-jar — APPROVED** [review]
version bump

**Pull request merged**